/*
 * Copyright 2020 The PyRec Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package pyrec.feature;

message IdWeight {
  oneof kind {
    bytes id_bytes = 1;
    int64 id_int = 2;
  }
  float weight = 3;
}

message BytesList {
  repeated bytes values = 1;
}
message FloatList {
  repeated float values = 1;
}
message IntList {
  repeated int64 values = 1;
}
message WeightedList {
  repeated IdWeight values = 1;
}

message FeatureList {
  oneof kind {
    BytesList bytes_list = 1;
    FloatList float_list = 2;
    IntList int_list = 3;
    WeightedList weighted_list = 4;
  }
}

message FeatureMap {
  map<int32, FeatureList> feature_map = 1;
}
